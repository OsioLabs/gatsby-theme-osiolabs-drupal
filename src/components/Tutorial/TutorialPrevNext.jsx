import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';

const TutorialPrevNext = ({ previous, next }) => (
  <div>
    {previous && <Link to={previous.path.alias}>&larr; {previous.title}</Link>}
    {next && <Link to={next.path.alias}>{next.title}&rarr;</Link>}
  </div>
);

TutorialPrevNext.propTypes = {
  previous: PropTypes.shape({
    title: PropTypes.string.isRequired,
    path: PropTypes.shape({
      alias: PropTypes.string.isRequired,
    }),
  }),
  next: PropTypes.shape({
    title: PropTypes.string.isRequired,
    path: PropTypes.shape({
      alias: PropTypes.string.isRequired,
    }),
  }),
};

export default TutorialPrevNext;
