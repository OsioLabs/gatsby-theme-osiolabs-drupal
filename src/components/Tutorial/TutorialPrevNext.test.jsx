import React from 'react';
import { render } from '@testing-library/react';
import 'jest-dom/extend-expect';

import TutorialPrevNext from './TutorialPrevNext';

const previous = {
  title: 'Previous tutorial',
  path: {
    alias: '/asdf/previous',
  },
};

const next = {
  title: 'Next tutorial',
  path: {
    alias: '/asdf/next',
  },
};

describe('Component: <TutorialPrevNext />', () => {
  it('renders correctly', () => {
    // Basic snapshot test.
    const { asFragment } = render(
      <TutorialPrevNext previous={previous} next={next} />
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });

  it('does not display previous link if no tutorial specified', () => {
    const { asFragment } = render(<TutorialPrevNext next={next} />);
    expect(asFragment().textContent).toContain('Next tutorial');
    expect(asFragment().textContent).not.toContain('Previous tutorial');
  });

  it('does not display next link if no tutorial specified', () => {
    const { asFragment } = render(<TutorialPrevNext previous={previous} />);
    expect(asFragment().textContent).not.toContain('Next tutorial');
    expect(asFragment().textContent).toContain('Previous tutorial');
  });
});
