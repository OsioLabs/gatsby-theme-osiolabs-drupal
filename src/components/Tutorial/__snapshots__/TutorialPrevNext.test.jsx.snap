// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <TutorialPrevNext /> renders correctly 1`] = `
<DocumentFragment>
  <div>
    <a
      href="/asdf/previous"
    >
      ← Previous tutorial
    </a>
    <a
      href="/asdf/next"
    >
      Next tutorial→
    </a>
  </div>
</DocumentFragment>
`;
